javaext/com.microsoft.java.lombok/src/main/java/com/microsoft/java/lombok/DataHandler.java

```diff
@@ -0,0 +1,101 @@
+        IType type = SourceAssistProcessor.getSelectionType(params, monitor);
+        if (type == null || type.getCompilationUnit() == null) {
+            return null;
+        }
+
+        TextEdit textEdit = new MultiTextEdit();
+        TextEdit allArgsConstructorTextEdit = ConstructorHandler.generateConstructor(params, monitor,
+                ConstructorKind.ALL_ARGS);
+        if (allArgsConstructorTextEdit != null) {
+            textEdit.addChild(allArgsConstructorTextEdit);
+        }
+        TextEdit noArgConstructorTextEdit = ConstructorHandler.generateConstructor(params, monitor,
+                ConstructorKind.NO_ARG);
+        if (noArgConstructorTextEdit != null) {
+            textEdit.addChild(noArgConstructorTextEdit);
+        }
+        TextEdit accessorsTextEdit = GetterSetterHandler.generateGetterSetter(params, monitor);
+        if (accessorsTextEdit != null) {
+            textEdit.addChild(accessorsTextEdit);
+        }
+        TextEdit toStringTextEdit = ToStringHandler.generateToString(params, monitor);
+        if (toStringTextEdit != null) {
+            textEdit.addChild(toStringTextEdit);
+        }
+        TextEdit hashCodeEqualsTextEdit = EqualsAndHashCodeHandler.generateHashCodeEquals(params, monitor);
+        if (hashCodeEqualsTextEdit != null) {
+            textEdit.addChild(hashCodeEqualsTextEdit);
+        }
+
+        return textEdit;
+    }
+
+    public static void removeMethods(IType type, ListRewrite rewriter, IProgressMonitor monitor) {
+        try {
+            CompilationUnit astRoot = CoreASTProvider.getInstance().getAST(type.getCompilationUnit(),
+                    CoreASTProvider.WAIT_YES, monitor);
+            if (astRoot == null) {
+                return;
+            }
+            ITypeBinding typeBinding = ASTNodes.getTypeBinding(astRoot, type);
+            if (typeBinding == null) {
+                return;
+            }
+            Set<String> dataMethods = new HashSet<String>(Arrays.asList(lombokCanEqualMethod, lombokEqualsMethod,
+                    lombokHashCodeMethod, lombokToStringMethod));
+            // Add constructors
+            dataMethods.add(typeBinding.getName());
+            // Add accessors
+            for (AccessorField accessor : GetterSetterHandler.getImplementedAccessors(type, AccessorKind.BOTH)) {
+                IField field = type.getField(accessor.fieldName);
+                dataMethods.add(GetterSetterUtil.getGetterName(field, null));
+                dataMethods.add(GetterSetterUtil.getSetterName(field, null));
+            }
+            IMethodBinding[] declaredMethods = typeBinding.getDeclaredMethods();
+            for (IMethodBinding item : declaredMethods) {
+                if (item.isDefaultConstructor()) {
+                    continue;
+                }
+                if (dataMethods.contains(item.getName())) {
+                    rewriter.remove(astRoot.findDeclaringNode(item), null);
+                }
+            }
```

Comment:
It's not accurate to remove methods based on method names.

Reply:
It seems that we can only know the lombok method's

- method name
- parameter names & types
- return type

Can we directly compare these things? Or is there any method to help us generate signature from them?

Reply:
Since you already get an `IMethodBinding` here, then try `new BindingKey(methodBinding.getKey()).toSignature()`.

Reply:
You can use the method name to fast check. If name matches, then check the full signature.